pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -0,0 +1,269 @@
+	"math/rand"
+	"net"
+	goruntime "runtime"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"k8s.io/klog"
+
+	"antrea.io/antrea/pkg/agent/flowexporter"
+	"antrea.io/antrea/pkg/agent/flowexporter/connections"
+	connectionstest "antrea.io/antrea/pkg/agent/flowexporter/connections/testing"
+	"antrea.io/antrea/pkg/agent/flowexporter/flowrecords"
+	interfacestoretest "antrea.io/antrea/pkg/agent/interfacestore/testing"
+	proxytest "antrea.io/antrea/pkg/agent/proxy/testing"
+	"antrea.io/antrea/pkg/ipfix"
+	queriertest "antrea.io/antrea/pkg/querier/testing"
+)
+
+const (
+	testPollInterval           = 2 * time.Second
+	testConnectionsCount       = 30000
+	testUpdateConnectionsCount = 20000
+	testNewConnectionsCount    = 4000
+	testDyingConnectionsCount  = 1000
+	testZoneFilter             = uint16(65520)
+	testDuration               = 15 * time.Second
+	testIdleTimeOut            = 5 * time.Second
+	testPerfActiveFlowTimeout  = 2 * time.Second
+	testPerfIdleFlowTimeout    = 1 * time.Second
+)
+
+var count = 0
+var exp *flowExporter
+
+func BenchmarkExport(b *testing.B) {
+	disableLogToStderr()
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+
+	registry := ipfix.NewIPFIXRegistry()
+	registry.LoadRegistry()
+	stopChan1 := make(chan struct{})
+	stopChan2 := make(chan struct{})
+	listener := startLocalServer(b, stopChan1, stopChan2)
+
+	conns := getConnections(testConnectionsCount)
+	records := flowrecords.NewFlowRecords()
+	mockIfaceStore := interfacestoretest.NewMockInterfaceStore(ctrl)
+	mockConnDumper := connectionstest.NewMockConnTrackDumper(ctrl)
+	mockProxier := proxytest.NewMockProxier(ctrl)
+	npQuerier := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+	connStore := connections.NewConnectionStore(mockConnDumper, records, mockIfaceStore, true, false, mockProxier, npQuerier, testPollInterval)
+	var err error
+	exp, err = NewFlowExporter(connStore, records, listener.String(), listener.Network(), testPerfActiveFlowTimeout, testPerfIdleFlowTimeout, false, true, false, nil, nil, false)
+	if err != nil {
+		b.Errorf("error in starting flow exporter: %v", err)
+	}
+	// Generate connections for Dumpflows, each round with update connections, new connections and dying connections
+	mockConnDumper.EXPECT().DumpFlows(testZoneFilter).Return(conns, testConnectionsCount, nil).Times(1)
+	for i := 0; i < 5; i++ {
+		conns = getUpdateConnections(conns, testNewConnectionsCount, testUpdateConnectionsCount, testDyingConnectionsCount, i)
+		mockConnDumper.EXPECT().DumpFlows(testZoneFilter).Return(conns, len(conns), nil).Times(1)
+	}
+	mockConnDumper.EXPECT().DumpFlows(testZoneFilter).Return(conns, len(conns), nil).AnyTimes()
+	mockConnDumper.EXPECT().GetMaxConnections().Return(testConnectionsCount, nil).AnyTimes()
+	mockIfaceStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(nil, false).AnyTimes()
+
+	var maxAlloc uint64
+	go statMaxMemAlloc(&maxAlloc, 500*time.Millisecond, stopChan2)
+	go connStore.Run(stopChan1)
+	go exp.Run(stopChan2)
```

Comment:
why separate channels?

Reply:
I want the poll to be finished after test duration and close exporter when exporter does not receive records for some time (`testIdleTimeOut`). That's why we need two separate channels to stop connection store and exporter, respectively.

Reply:
>... close exporter when exporter does not receive records for some time (testIdleTimeOut)

Do you mean the collector could not receive the records? The exporter just sends the records in the test here.
Any particular reason to stop polling? 
In other words, do we want to wait till we receive all messages at the collector? Could we just do the duration and close the collector/exporting processes when it finishes? We just record the number of messages at the collector before closing the process. 
This way we can see within specific duration, how many messages are collected. As CPU time is optimized, we could see an increase in the number of messages at the collector. What do you think of this approach?

Reply:
Got it. Yes this approach makes sense to me. As you mentioned, I was planning to wait for all messages to safely close the exporting process.